subsys/bluetooth/host/conn.c

```diff
@@ -671,6 +671,37 @@ static bool send_buf(struct bt_conn *conn, struct net_buf *buf)
 static struct k_poll_signal conn_change =
 		K_POLL_SIGNAL_INITIALIZER(conn_change);
 
+static void conn_destroy(struct bt_conn *conn, void *data)
+{
+	struct net_buf *buf;
+
+	/* Give back any allocated buffers */
+	while ((buf = net_buf_get(&conn->tx_queue, K_NO_WAIT))) {
+		struct bt_conn_tx *tx = tx_data(buf)->tx;
+
+		tx_data(buf)->tx = NULL;
+
+		/* destroy the buffer */
+		net_buf_unref(buf);
+
+		/* destroy the tx context (and any associated meta-data) */
+		if (tx) {
+			conn_tx_destroy(conn, tx);
+		}
+	}
+
+	bt_conn_reset_rx_state(conn);
+
+	k_work_reschedule(&conn->deferred_work, K_NO_WAIT);
```

Comment:
Does it matter if `k_work_reschedule` returns an error here? If no, can you add a comment in the code explaining that we are ignoring the errors on purpose (and why it's not a problem)? Same questions for the addition of `k_work_cancel_delayable` further down.

Reply:
this seems super similar to `conn_cleanup(struct bt_conn *conn)`. I think this fn can just be a call to that + the `set_state` after.